shapeshift/web
# fix: handle pending cached queries & no-op inputs

## Description

Fixes 2 bugs:

- A race condition where cached, pending queries would be ignored
- A no-op on input would clear the receive amounts

## Pull Request Type


## Issue (if applicable)

N/A

## Risk

Small

## Testing

To test the race condition bug is fixed:

1. Enter an amount, say '1'
2. Quickly enter add another input number, say '2'
3. Quickly remove the '2' so we are back to one

Previously a quote would come back for '12', and we would not update the quote to reflect the current value ('1') in this example

This should now be fixed.

To test the no-op issue is fixed:

1. Enter trade amount and see that the values are populated
2. Do a no-op input (e.g. add a symbol to the end of the input (e.g. '/', or enter the exact same input amount by pasting over top)
3. Ensure the receive amount stays populated

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

N/A
